File: CaesarCipher/Python/caesar_cipher.py
# lowercase a in ASCII corresponds to 61
# span of lowercase letters 61 - (61+18)
# uppercase A corresponds to 41

# naive solution
def caesar_cipher(plaintext, key):
    cipher_text = []
    for char in plaintext:
        if char.isalpha():
            if char.islower():
                char = chr(ord(char) + key) if ord(char) + key <= 122 else chr(ord(char) + key - 26)
            elif char.isupper():
                char = chr(ord(char) + key) if ord(char) + key <= 90 else chr(ord(char) + key - 26)
        cipher_text.append(char)
    
    return "".join(cipher_text)


if __name__ == "__main__":
    text = "My very secret message"
    key = -15
    encrypted_text = caesar_cipher(text, key)
    print(encrypted_text)

File: CaesarCipher/Java/CaesarCipher.java
public class CaesarCipher {
    public static void main(String[] args) {
        String cipherText = CaesarCipher.encrypt("My very secret message...", 1);
        System.out.println(cipherText);
    }

    public static String encrypt(String plainText, int key) {
        StringBuilder cipherText = new StringBuilder();
        for (char letter : plainText.toCharArray()) {
            if (Character.isLetter(letter)) {
                char base = Character.isUpperCase(letter) ? 'A' : 'a';
                cipherText.append((char) (((letter - base + key) % 26) + base)); // smart way to 'wray around' ASCII letters
            } else {
                cipherText.append(letter);
            }
        }

        return cipherText.toString();
    }
} 
